roku/device/roku-tv.groovy

```diff
@@ -321,21 +324,25 @@ void scheduleQueryActiveApp() {
 }
 
 void scheduleQueryMediaPlayer() {
-    unschedule('queryMediaPlayer')
+    if (this[SETTING_LOG_ENABLE]) { log.info "scheduleQueryMediaPlayer: Executed " }
+//    unschedule('queryMediaPlayer')
 
     Long delay = (this[SETTING_MEDIA_INTERVAL] ?: 0) * (this[SETTING_MEDIA_UNITS] == REFRESH_UNIT_MINUTES ? 60 : 1)
 
-    if (device.currentValue('application', true) == 'Roku') {
+    if (device.latestValue('application', true) == 'Roku') {
+        if (this[SETTING_LOG_ENABLE]) { log.info "scheduleQueryMediaPlayer: App Compare is Roku" }
         return
     }
 
     if (this[SETTING_DEVICE_IP] && delay > 0) {
+        if (this[SETTING_LOG_ENABLE]) { log.info "scheduleQueryMediaPlayer: App Compare is not Roku schedule next check" }
         runIn(delay, 'queryMediaPlayer')
     }
 }
 
 void scheduleQueryInstalledApps() {
-    unschedule('queryInstalledApps')
+    if (this[SETTING_LOG_ENABLE]) { log.info "scheduleQueryInstalledApps: Executed " }
+//    unschedule('queryInstalledApps')
```

Comment:
Can you do a performance comparison of the extra unschedule calls vs not having them?  The impact should be negligible.

Reply:
Ok so I have done some testing.

Device: Roku Streaming Stick
Device Refresh : 0 - Disabled
Active App Refresh : 10 Seconds
Media Player Refresh : 3 Seconds
Installed Apps Refresh: 0 - Disabled

First test was to validate the impact of the Unschedule for the Active App and Media Player. 
Original driver code with Unschedule in place used 1.53% cpu time over a 30 min interval
New driver code with unscheduled removed used 1.26% cpu time over a 30 min interval
Difference of 0.27% cpu usage.

Second Test was to validate the impact of Unschedule as part of the Active App Query running. This was to validate if the new routine for setCurrentApplication that eliminated the unschedule that occured each time it ran and the current app was roku made a difference.

Old code used .44% over a 34min interval
new code used .33% over a 34min interval.

So no doubt that both of these things increased CPU Usage 

One other metric I wanted to through out here is that my Actual RokuTV with the new code is showing 0.07% cpu usage. over the 34 min interval of the last test. The key thing to recognize with it is that it is running the queryDeviceInfo only since the updated code unschedules all of the others. It is only running once a min since it is being used for Power only and the extra unschedule is removed.

Let me know if there is any other test condition you want me to check.

Reply:
I know these differences don't seem like much but when all other Zigbee/zwave devices use so much less resources any room for improvement on the WIFI based devices will help  to maximize ability for a higher number of devices on the hub.

Reply:
I am totally on board with find ways to squeeze all the performance out of this as possible.  I just don't want to lose functionality in doing so.  All the LAN based devices that are not made for home automation (meaning they don't have a an event notification process built in) are high CPU usage devices.  This is in part because of all the parsing of verbose messages, and unnecessary queries that result in no net change.  But I believe it is also because the hardware or software is not optimized for working with multiple concurrent network streams.  I believe the event bus methodology of the ST / HE systems is serialized, resulting in sequential processing of events, rather than parallel processing.  And this can cause the system to perform really slow, even if the cpu is under low load.    And the schedule/unschedule I would include as serialized processes that only happen one at a time.  All this is just a hypothesis based on observations, but w/o any good way to test.
So, all this input is very helpful.  And having more than just myself testing ensures more entropy in the test which helps make this more reliable in the end.